src/main.rs

```diff
@@ -172,11 +203,49 @@ impl FromStr for Blake3Cid {
     }
 }
 
+async fn make_rpc_client(
+    rpc_port: u16,
+) -> anyhow::Result<RpcClient<ProviderService, QuinnConnection<ProviderResponse, ProviderRequest>>>
+{
+    let endpoint = iroh::get::make_client_endpoint(None, vec![RPC_ALPN.to_vec()], false, false)?;
+    let addr = SocketAddr::new(Ipv4Addr::LOCALHOST.into(), rpc_port);
+    let server_name = "localhost".to_string();
+    let connection = QuinnConnection::new(endpoint, addr, server_name);
+    let client = RpcClient::<ProviderService, _>::new(connection);
+    // Do a version request to check if the server is running.
+    let _version = tokio::time::timeout(Duration::from_secs(1), client.rpc(VersionRequest))
+        .await
+        .context("iroh server is not running")??;
+    Ok(client)
+}
+
 const PROGRESS_STYLE: &str =
     "{msg}\n{spinner:.green} [{elapsed_precise}] [{wide_bar:.cyan/blue}] {bytes}/{total_bytes} ({eta})";
 
-#[tokio::main(flavor = "multi_thread")]
-async fn main() -> Result<()> {
+/// drop a RpcClient and wait for a short time.
+///
+/// The waiting is not strictly necessary, but if the program terminates immediately
+/// after closing the client, the close message is not being sent to the server,
+/// and the server will hold on to the connection until it times out.
+async fn close<C>(client: RpcClient<ProviderService, C>) -> anyhow::Result<()> {
+    drop(client);
+    tokio::time::sleep(Duration::from_millis(100)).await;
```

Comment:
Yes, this would have to be a tokio task though because it is called from drop. Then I guess it would be just another background task that gets some grace period to execute.

Let's fix it in another PR though. This works.